src/d5a50_len_5fd0.c

```diff
@@ -68,13 +70,40 @@ void draw_hud_element_clipped(s32 arg0) {
     draw_hud_element(arg0, 0);
 }
 
-INCLUDE_ASM(s32, "d5a50_len_5fd0", draw_hud_element_2);
+void draw_hud_element_2(s32 arg0) {
+    draw_hud_element(arg0, 1);
+}
 
 void draw_icon_2(s32 iconID) {
     draw_hud_element(iconID, 2);
 }
 
-INCLUDE_ASM(s32, "d5a50_len_5fd0", set_hud_element_script);
+void set_hud_element_script(s32 arg0, s32 *arg1) {
+    HudElement *hudElement;
+
+    hudElement = gHudElementList[arg0 & ~0x800];
+    if (arg1 == NULL) {
+        arg1 = D_8014EFC8;
+    }
+
+    hudElement->updateTimer = 1;
+    hudElement->widthScale = 1024;
+    hudElement->heightScale = 1024;
+    hudElement->readPos = arg1;
+    hudElement->startReadPos = arg1;
+    hudElement->ptrPropertyList = arg1;
+    hudElement->screenPosOffset[0] = 0;
+    hudElement->screenPosOffset[1] = 0;
+    hudElement->worldPosOffset[0] = 0;
+    hudElement->worldPosOffset[1] = 0;
+    hudElement->flags &= ~4;
+    hudElement->uniformScale = 1.0f;
+    hudElement->flags &= ~0x930;
+    load_hud_element(hudElement, arg1);
+    
+    while (hud_element_update(hudElement) != 0) {}
```

Comment:
we may want to write this differently, like as an assert or something. I'll let alex weigh in

Reply:
I don't think an assert would work here, because the function gets called for each loop iteration. I'm good with how it is